flink-core/src/main/java/org/apache/flink/api/java/typeutils/TypeExtractor.java

```diff
@@ -1713,6 +1714,12 @@ private static Type materializeTypeVariable(ArrayList<Type> typeHierarchy, TypeV
 			return timeTypeInfo;
 		}
 
+		// check for SQL time types
+		TypeInformation<OUT> localTimeTypeInfo = LocalTimeTypeInfo.getInfoFor(clazz);
```

Comment:
Changes in the type extractor lead to breaking backwards compatibility. So far we don't have a migration path from one serializer to another. Right, @tzulitai?

Reply:
Yeah, `GenericTypeInfo` will change to `LocalTimeTypeInfo`...

Reply:
I'll remove it.